### PR TITLE
ocamlPackages.fiber: unstable-2023-02-28 -> 3.7.0

### PR DESCRIPTION
**Pull request by marsam**
https://github.com/ocaml-dune/fiber/releases/tag/3.7.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/fiber/default.nix [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
-    owner = "ocaml-dune";
-    repo = "fiber";
-    rev = "5563b588c1313f128eafa74d66f0626c9128d34d";
-    hash = "sha256-18GfGXpu+uiIiCuLhIx5z5jRkem1nNWaQB6Ms0AE9sE=";
+  src = fetchurl {
+    url = "https://github.com/ocaml-dune/fiber/releases/download/${version}/fiber-lwt-${version}.tbz";
+    hash = "sha256-hkihWuk/5pQpmc42iHQpo5E7YoKcRxTlIMwOehw7loI=";
   };
 
   duneVersion = "3";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/fiber/default.nix:12]: This line is no longer needed and may be removed.

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocaml-ng.ocamlPackages_5_1.fiber

**Comment by vbgl** [comment]
Tests tend to fail. It seems to be an issue with the test-suite, though.

~~~
--- fiber/test/fiber_tests.ml   2023-11-17 20:21:34.267584276 +0000
+++ fiber/test/fiber_tests.ml.corrected 2023-11-17 20:21:35.627614126 +0000
@@ -675,7 +675,7 @@
   (* Check that the number of live words keeps on increasing because we are
      leaking memory: *)
   test ~pred:( < ) ~iter_function:naive_stream_parallel_iter;
-  [%expect {| [ 52; 115; 745; 7045 ] |}];
+  [%expect {| [ 48; 111; 741; 7041 ] |}];
   test ~pred:( = ) ~iter_function:Fiber.Stream.In.parallel_iter;
   [%expect {| [ 43; 43; 43; 43 ] |}]
 ~~~

**Comment by ulrikstrid** [comment]
@Alizter do you know anything about the tests?

**Comment by Alizter** [comment]
i'm not sure why these thests fail, but most likely it is tk do with OCaml 5.1.  I would just disable running the tests on versions later than 4.14. cc @rgrinberg so he knows about this issue too.

**Comment by marsam** [comment]
I've disabled the tests for OCaml>=5

**Comment by Alizter** [comment]
Another alternative would be to add a patch to fix the test. It seems that the memory layout changed in OCaml 5.1 causing this test, which is counting some bytes, to have slightly altered values.

**Comment by vbgl** [comment]
Tests also fail for OCaml < 4.13.

**Comment by Alizter** [comment]
I would suggest disabling for anything other than 4.14 then. I've made a suggestion in https://github.com/ocaml-dune/fiber/issues/27 to possibly make this version agnostic upstream.

**Comment by vbgl** [comment]
This looks like the correct fix, from the packaging side of things.

**Pull request merged**